functional/server_test.go

```diff
@@ -41,34 +41,60 @@ func TestReconfigureServer(t *testing.T) {
 		t.Fatal(err)
 	}
 
-	// NOTE: we need to sleep once here to get reliable results.
-	// Without this sleep, the entire fleetd test always ends up succeeding
-	// no matter whether SIGHUP came or not.
-	_, _ = cluster.MemberCommand(m0, "sh", "-c", `'sleep 2'`)
-
 	err = waitForFleetdSocket(cluster, m0)
 	if err != nil {
 		t.Fatalf("Failed to get a list of fleetd sockets: %v", err)
 	}
 
-	// send a SIGHUP to fleetd, and periodically checks if a message
+	unit := fmt.Sprintf("fixtures/units/hello.service")
+	stdout, stderr, err := cluster.Fleetctl(m0, "start", unit)
+	if err != nil {
+		t.Fatalf("Failed starting unit: \nstdout: %s\nstderr: %s\nerr: %v", stdout, stderr, err)
+	}
+
+	_, err = cluster.WaitForNActiveUnits(m0, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Trigger AgentReconciler just here
+	stdout, stderr, err = cluster.Fleetctl(m0, "unload", unit)
+	if err != nil {
+		t.Fatalf("Failed unloading unit: \nstdout: %s\nstderr: %s\nerr: %v", stdout, stderr, err)
+	}
+
+	// Send a SIGHUP to fleetd, and periodically checks if a message
 	// "Reloading configuration" appears in fleet's journal, up to timeout (15) seconds.
-	stdout, _ := cluster.MemberCommand(m0, "sudo", "systemctl", "kill", "-s", "SIGHUP", "fleet")
+	stdout, _ = cluster.MemberCommand(m0, "sudo", "systemctl", "kill", "-s", "SIGHUP", "fleet")
 	if strings.TrimSpace(stdout) != "" {
 		t.Fatalf("Sending SIGHUP to fleetd returned: %s", stdout)
 	}
 
-	err = waitForReloadConfig(cluster, m0)
-	if err != nil {
-		t.Fatalf("Failed to get log about reconfiguration: %v", err)
+	// Watch the logs if fleet was correctly reloaded
+	errSigHup := waitForReloadConfig(cluster, m0)
+	if errSigHup != nil {
+		t.Logf("Failed to ensure that fleet was correctly reloaded: %v", errSigHup)
 	}
 
 	// check if fleetd is still running correctly, by running fleetctl status
-	_, _, err = cluster.Fleetctl(m0, "list-units")
+	// Even if the log message do not show up this test may catch the error.
+	stdout, _, err = cluster.Fleetctl(m0, "list-units")
 	if err != nil {
 		t.Fatalf("Unable to check list-units. Please check for fleetd socket. err:%v", err)
 	}
 
+	// Ensure that fleet received SIGHUP, if not then just skip this test
+	// probably due to journald and or other delays.
+	if errSigHup != nil {
+		err = waitForReloadConfig(cluster, m0)
```

Comment:
I don't understand what the point is in checking this again here? If it failed earlier, this means the test wouldn't produce any useful results anyway I think?...